front/pages/w/[wId]/builder/data-sources/managed.tsx

```diff
@@ -328,6 +412,38 @@ export default function DataSourcesView({
         ) : (
           false
         )}
+        {selectedDataSource && selectedDataSource.connector && (
+          <>
+            <ConnectorPermissionsModal
+              owner={owner}
+              connector={selectedDataSource.connector}
+              dataSource={selectedDataSource}
+              isOpen={showConnectorModal && !!selectedDataSource}
+              onClose={() => {
+                setShowConnectorModal(false);
+              }}
+              setShowEditionModal={setShowEditionModal}
+              handleUpdatePermissions={handleUpdatePermissions}
+            />
+            <DataSourceEditionModal
+              isOpen={showEditionModal}
+              onClose={() => setShowEditionModal(false)}
+              dataSource={selectedDataSource}
+              owner={owner}
+              user={user}
+              onEditPermissionsClick={() => {
+                if (!selectedDataSource.connector) {
```

Comment:
This is unfortunately needed to please the compiler which doesn't infer it from:
```
{selectedDataSource && selectedDataSource.connector && (
```
🤷🏼‍♂️